src/Executor/Setup.cs

```
@@ -0,0 +1,290 @@
+                }
+            }
+            return commandSetting;
+        }
+
+        private bool BuildRequiredValueSettingsForCommand(Dictionary<string, string> requiredSettings, Dictionary<string, string> commandValues)
+        {
+            foreach (KeyValuePair<string, string> reqSetting in requiredSettings)
+            {
+                string value = string.IsNullOrEmpty(reqSetting.Value) || reqSetting.Value.Equals("default") ? FindSettingValue(reqSetting.Key) : reqSetting.Value;
+                if (value != null && (string.IsNullOrEmpty(commandValues[reqSetting.Key]) || reqSetting.Key.Equals("Project")))
+                {
+                    commandValues[reqSetting.Key] = string.IsNullOrEmpty(value) ? "True" : value;
+                }
+                else
+                {
+                    if (!string.IsNullOrEmpty(value) && !value.Equals(commandValues[reqSetting.Key]))
+                    {
+                        Console.WriteLine("Error: The value for setting {0} can't be overwriten.", reqSetting.Key);
+                        return false;
+                    }
+                }
+            }
+            return true;
+        }
+
+        private bool BuildOptionalValueSettingsForCommand(Dictionary<string, string> optionalSettings, Dictionary<string, string> commandValues)
+        {
+            foreach (KeyValuePair<string, string> optSetting in optionalSettings)
+            {
+                if (string.IsNullOrEmpty(commandValues[optSetting.Key]))
+                {
+                    string value = string.IsNullOrEmpty(optSetting.Value) || optSetting.Value.Equals("default") ? FindSettingValue(optSetting.Key) : optSetting.Value;
+                    if (value != null)
+                    {
+                        commandValues[optSetting.Key] = value;
+                    }
+                    else
+                    {
+                        return false;
```

This one doesn't Console.WriteLine anything... if this fails, does the process just silently do nothing?